### PR TITLE
Refactor code to improve readability

### PR DESCRIPTION
**Pull request by vastamaki**
- Add new filter option
- Remove duplicate css/code
- Update component styles
- Use current user teams as default teams
- Refactor some parts of the code
- Update readme

### CODE REVIEW

File: src/Components/Auth/index.js [hunk 1]
```diff
@@ -20,24 +21,20 @@ const checkToken = async (rest) => {
...
-      response = await fetch(
-        encodeURI(
-          `https://app.pagerduty.com/oauth/token?grant_type=authorization_code&client_id=ba65171a721befb7fc2b3ceece703a6b38c1da83c14954039f81a7115bb2058e&redirect_uri=http://localhost:3000&code=${authorizationCode}&code_verifier`,
-        ),
-        params,
-      );
-    } catch (error) {
-      return false;
-    }
+    const response = await fetch(
```
Comment (michalksiezuk-nc) [inline src/Components/Auth/index.js]: Awaited `fetch()` call should be wrapped in `try/catch`.

Comment (michalksiezuk-nc) [reply src/Components/Auth/index.js]: We should also think about getting the environment properly, hard-coded return URL is good for now though.

File: src/Components/Incidents/index.js [hunk 2]
```diff
@@ -53,10 +53,23 @@ class Incidents extends PureComponent {
...
+      const excluded = filters.exclude
+        .split(',')
+        .some((filter) => incident.service.summary.includes(filter));
+      if (excluded) return false;
+    }
+    if (
+      filters.showOnlyOwnIncidents
+      && incident.last_status_change_by.summary !== this.context.currentUser.name
+    ) return false;
+    return true;
```
Comment (michalksiezuk-nc) [inline src/Components/Incidents/index.js]: You could simplify this one:

```suggestion
    return !(filters.showOnlyOwnIncidents
      && incident.last_status_change_by.summary !== this.context.currentUser.name);
```

File: src/Components/Incidents/index.js [hunk 3]
```diff
@@ -74,66 +87,61 @@ class Incidents extends PureComponent {
...
+                        >
+                          <h3 className="summary">
+                            <p
+                              style={{
+                                backgroundColor: this.incidentStatusToColor(
+                                  incident,
+                                ),
+                              }}
+                              className="incident-status"
+                            ></p>
```
Comment (michalksiezuk-nc) [inline src/Components/Incidents/index.js]: As `<p>` tag is empty you could use XML-style closing `<p prop="val" />`.

File: src/Components/Incidents/index.js [hunk 4]
```diff
@@ -53,10 +53,23 @@ class Incidents extends PureComponent {
...
+      if (excluded) return false;
+    }
+    if (
+      filters.showOnlyOwnIncidents
+      && incident.last_status_change_by.summary !== this.context.currentUser.name
+    ) return false;
+    return true;
+  };
+
   render() {
```
Comment (michalksiezuk-nc) [inline src/Components/Incidents/index.js:119]: `render()` became really long and complex. I suggest refactoring it a bit by extracting more complex parts (like L107-L114) to separate render methods.

File: src/Components/Modals/Cards/index.js [hunk 5]
```diff
@@ -45,67 +44,65 @@ class Cards extends PureComponent {
...
-        </div>
-      </div>
+      <React.Fragment>
+        <h2>Card content</h2>
+        <ul>
+          <li>
+            <p>
+              <input
+                type="checkbox"
+                id="switch1"
```
Comment (michalksiezuk-nc) [inline src/Components/Modals/Cards/index.js:54]: Using more descriptive names is highly recommended :slightly_smiling_face:  These are related to things that are named already, so I would use that, e.g. `summary-checkbox`.

File: src/Components/Modals/Filters/index.js [hunk 6]
```diff
@@ -39,24 +48,39 @@ class Filters extends PureComponent {
...
+              value={this.state.filters.exclude}
+            />
+          </li>
+          <li>
+            <p>
+              <input
+                type="checkbox"
+                id="switch1"
+                onChange={(e) => this.handleCheckboxChange(e, 'showOnlyOwnIncidents')
+                }
```
Comment (michalksiezuk-nc) [inline src/Components/Modals/Filters/index.js]: This closing brace belongs in L68 :slightly_smiling_face:

File: src/Components/Modals/Settings/index.js [hunk 7]
```diff
@@ -1,39 +1,34 @@
...
-            />
-          </div>
-        </div>
+        <h2>Settings</h2>
+        <input
+          onClick={() => changeModalState({
+            modal: 'teams',
+            state: true,
+          })(dispatch)
+          }
```
Comment (michalksiezuk-nc) [inline src/Components/Modals/Settings/index.js]: Formatting looks a bit weird. This brace could be move to line above.

File: src/Components/Modals/Settings/index.js [hunk 8]
```diff
@@ -1,39 +1,34 @@
...
+          className="submit"
+          type="submit"
+          value="Select team"
+        />
+        <input
+          onClick={() => changeModalState({
+            modal: 'cardSettings',
+            state: true,
+          })(dispatch)
+          }
```
Comment (michalksiezuk-nc) [inline src/Components/Modals/Settings/index.js]: Formatting issue, as in L17.

File: src/Components/Modals/index.css [hunk 9]
```diff
@@ -28,18 +28,24 @@ h3 {
   min-width: 300px;
 }
 
-.card-settings > ul {
+.modal > h2 {
+  margin: 0px;
```
Comment (michalksiezuk-nc) [inline src/Components/Modals/index.css]: Good practice, 0-value properties does not require the unit.

File: src/Components/Modals/index.js [hunk 10]
```diff
@@ -5,26 +5,22 @@ import Filters from './Filters';
...
-      return <Filters />;
-    }
-    if (openModals.sorting) {
-      return <Sorting />;
-    }
-    if (openModals.settings) {
-      return <Settings />;
-    }
-    return null;
+    return Object.values(openModals).some((e) => e === true) && (
```
Comment (michalksiezuk-nc) [inline src/Components/Modals/index.js]: Much better solution, I would just name `e` in some more descriptive fashion.

File: src/Context/actions.js [hunk 11]
```diff
@@ -8,33 +10,60 @@ export const getTeams = () => async (dispatch) => {
...
+export const getTeams = () => async (dispatch) => {
+  const params = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/vnd.pagerduty+json;version=2',
+      Authorization: `Bearer ${localStorage.getItem('access_token')}`,
+    },
+  };
+
+  let response;
```
Comment (michalksiezuk-nc) [inline src/Context/actions.js]: I suggest rewriting this whole part (L34-L46) like this:

```js
  try {
    const response = await fetch(
      encodeURI('https://api.pagerduty.com/teams'),
      params,
    );

    dispatch({
      type: 'GET_TEAMS',
      payload: response.teams,
    });
  } catch (err) {
    throw new Error(err);
  }
```

File: src/Context/actions.js [hunk 12]
```diff
@@ -8,33 +10,60 @@ export const getTeams = () => async (dispatch) => {
   };
 
   const response = await fetch(
-    encodeURI('https://api.pagerduty.com/teams'),
+    encodeURI('https://api.pagerduty.com/users/me'),
```
Comment (michalksiezuk-nc) [inline src/Context/actions.js]: This request should be wrapped in try/catch, similar to how `getTeams()` handles it.

File: src/Context/index.js [hunk 13]
```diff
@@ -105,14 +122,18 @@ export class Provider extends Component {
...
     dispatch: (action) => this.setState((state) => reducer(state, action)),
   };
 
-  componentDidMount = () => {
+  componentDidMount = async () => {
     const filters = JSON.parse(localStorage.getItem('filters'));
     const hoursMarked = JSON.parse(localStorage.getItem('hoursMarked'));
     const cardContent = JSON.parse(localStorage.getItem('cardContent'));
     const sortBy = JSON.parse(localStorage.getItem('sortBy')) || {};
+    await setCurrentUser()(this.state.dispatch);
```
Comment (michalksiezuk-nc) [inline src/Context/index.js]: Those awaits deserve try/catch :slightly_smiling_face:

File: src/GetView.js [hunk 14]
```diff
@@ -40,65 +34,43 @@ export default class GetView extends PureComponent {
       },
     };
 
+    const incidents = [];
+    let offset = 0;
     let response;
-
-    try {
+    do {
+      /* eslint-disable no-await-in-loop */
```
Comment (michalksiezuk-nc) [inline src/GetView.js:41]: This is a really bad practice, please have a look [here](https://eslint.org/docs/rules/no-await-in-loop) and refactor to use `Promise.all()`.

File: src/index.css [hunk 15]
```diff
@@ -39,12 +39,11 @@ body,
   color: #1d1d1d;
   text-align: center;
   font-size: 22px;
-  margin: 10px;
+  margin: 10px 0px 10px 0px;
```
Comment (michalksiezuk-nc) [inline src/index.css]: 0 value does not require a unit

**Review by michalksiezuk-nc — APPROVED** [review]

File: src/GetView.js (hunk 14 above)
Comment (vastamaki) [reply src/GetView.js:41]: I know, but in cases like this (when we need data from the response to end loop) this is fine as far as I know. If there is any other solution, please let me know.

**Pull request merged**